smartsim/_core/mli/comm/channel/dragonchannel.py

```diff
@@ -43,24 +43,59 @@
 class DragonCommChannel(cch.CommChannelBase):
     """Passes messages by writing to a Dragon channel"""
 
-    def __init__(self, key: bytes) -> None:
-        """Initialize the DragonCommChannel instance"""
-        super().__init__(key)
-        self._channel: dch.Channel = dch.Channel.attach(key)
+    def __init__(self, channel: "dch.Channel", recv_timeout: int = 0) -> None:
+        """Initialize the DragonCommChannel instance
+
+        :param channel: a channel to use for communications
+        :param recv_timeout: a default timeout to apply to receive calls"""
+        serialized_ch = channel.serialize()
+        safe_descriptor = base64.b64encode(serialized_ch).decode("utf-8")
+        super().__init__(safe_descriptor)
+        self._channel = channel
+        self._recv_timeout = recv_timeout
+
+    @property
+    def recv_timeout(self) -> int:
+        """The timeout for receive requests (in seconds)"""
+        return self._recv_timeout
 
     def send(self, value: bytes) -> None:
         """Send a message throuh the underlying communication channel
         :param value: The value to send"""
         with self._channel.sendh(timeout=None) as sendh:
             sendh.send_bytes(value)
 
-    def recv(self) -> t.List[bytes]:
+    def recv(self, timeout: int = 0) -> t.List[bytes]:
         """Receieve a message through the underlying communication channel
 
+        :param timeout: maximum time to wait for messages to arrive
         :returns: the received message"""
-        with self._channel.recvh(timeout=None) as recvh:
-            message_bytes: bytes = recvh.recv_bytes(timeout=None)
-            return [message_bytes]
+        timeout = timeout or self._recv_timeout
+        with self._channel.recvh(timeout=timeout) as recvh:
+            messages: t.List[bytes] = []
+
+            # todo: consider that this could (under load) never exit. do we need
+            # to configure a maximum number to pull at once?
```

Comment:
Hmmmm didn't think about this. We'd want all of the messages within the channel though, right?

Reply:
i think it's possible with a while to never run out of new messages to receive. it doesn't event need to be a ton of buffered messages, just constant. 

i think a `for i in range(max_message_retrievals)` may ensure that it doesn't get stuck and any loop triggering retrieval would end up getting called again, anyway...

Reply:
- Completed extraction of the looping within a channel. 
- Updated tests as necessary.
- Implemented a batch timeout loop in the `EventConsumer` and added tests.